src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/aarch64/CallArranger.java

```diff
@@ -200,13 +201,13 @@ static boolean isHomogeneousFloatAggregate(MemoryLayout type) {
         if (!(baseType instanceof ValueLayout))
             return false;
 
-        var optAbiType = baseType.abiType();
+        var optAbiType = baseType.attribute(SystemABI.NATIVE_TYPE, SystemABI.Type.class);
```

Comment:
Since all ABI layouts are going to have an annotation of this kind, would it make sense to add a static helper function in the SystemABI.Type enum to get the ABI type from a layout e.g.

```
Type fromLayout(Layout layout) throws IllegalArgumentException //if not an ABI layout
```

Reply:
Good idea, will add this as well.